api/v1_projects.inc

```diff
@@ -410,12 +415,15 @@ function api_v1_project_page_round($method, $data, $query_params)
             state
         FROM %s
         WHERE image = '%s'
-    ", $text_column, $user_column, $data[":projectid"]->projectid,
+    ", $text_column, $user_column, $project->projectid,
         DPDatabase::escape($data[":pagename"])
     );
     $result = DPDatabase::query($sql);
     $row = mysqli_fetch_assoc($result);
-    $row["image_url"] = $data[":projectid"]->url . "/" . $row["image"];
+    if (!$row) {
+        throw new NotFoundError("No such page in project", 1006);
```

Comment:
We should either treat these strings as canonical, or the numbers as canonical, but not both. That is, if we want to use unique error codes like this we need to define them somewhere with the strings they belong to and treat them as constants in the code. We then need an API endpoint to return them to the API developer so they don't need to slog through the code to see what each one is.

Reply:
Yes, I was thinking the same.